### DISCUSSION TITLE
It seems `Stream` is not impl for `Interval` in tokio 1.0 ?

### DISCUSSION THREAD

**Discussion by valbendan** [tokio-rs/tokio#3341] [Q&A] [answered]
I already have add `tokio_stream` to my dep

**Comment by Darksonn** [comment, marked as answer]
The `tokio-stream` crate cannot add an impl of `Stream` to `tokio::time::Interval` because neither the trait nor the type is defined in `tokio-stream`. Instead you can use the `async-stream` crate to turn it into a stream.
```Rust
let stream = async_stream::stream! {
    loop {
        yield interval.tick().await;
    }
};
```
You can also implement `Stream` on your wrapper type like this:
```Rust
use tokio::time::{Interval, Instant};
use std::pin::Pin;
use std::task::{Context, Poll};
use tokio_stream::Stream;

struct MyInterval {
    int: Interval,
}

impl Stream for MyInterval {
    type Item = Instant;

    fn poll_next(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Instant>> {
        self.int.poll_tick(cx).map(|inner| Some(inner))
    }
}
```
The `tokio-stream` crate should soon receive [an update][1] that makes this easier.

[1]: https://github.com/tokio-rs/tokio/pull/3343
